CRM/Case/Form/Task/PickProfile.php

```diff
@@ -0,0 +1,148 @@
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2017
+ * $Id$
+ *
+ */
+
+/**
+ * This class provides the functionality for batch profile update for case
+ */
+class CRM_Case_Form_Task_PickProfile extends CRM_Case_Form_Task {
+
+  /**
+   * The title of the group
+   *
+   * @var string
+   */
+  protected $_title;
+
+  /**
+   * Maximum cases that should be allowed to update
+   */
+  protected $_maxCases = 100;
+
+  /**
+   * Variable to store redirect path
+   */
+  protected $_userContext;
+
+  /**
+   * Build all the data structures needed to build the form.
+   *
+   * @return void
+   */
+  public function preProcess() {
+    // initialize the task and row fields
+    parent::preProcess();
+    $session = CRM_Core_Session::singleton();
+    $this->_userContext = $session->readUserContext();
+
+    CRM_Utils_System::setTitle(ts('Update multiple cases'));
+
+    $validate = FALSE;
+    //validations
+    if (count($this->_caseIds) > $this->_maxCases) {
+      CRM_Core_Session::setStatus(ts("The maximum number of cases you can select for Update multiple cases is %1. You have selected %2. Please select fewer cases from your search results and try again.", array(
+        1 => $this->_maxCases,
+        2 => count($this->_caseIds),
+      )), ts('Update multiple records error'), 'error');
+      $validate = TRUE;
+    }
+
+    // than redirect
+    if ($validate) {
+      CRM_Utils_System::redirect($this->_userContext);
+    }
+  }
+
+  /**
+   * Build the form object.
+   *
+   *
+   * @return void
+   */
+  public function buildQuickForm() {
+    $types = array('Case');
+    $profiles = CRM_Core_BAO_UFGroup::getProfiles($types, TRUE);
+
+    if (empty($profiles)) {
+      CRM_Core_Session::setStatus(ts("You will need to create a Profile containing the %1 fields you want to edit before you can use Update multiple cases. Navigate to Administer CiviCRM >> CiviCRM Profile to configure a Profile. Consult the online Administrator documentation for more information.", array(1 => $types[0])), ts('Update multiple records error'), 'error');
```

Comment:
This information is not up to date, when I tried this PR branch I needed to go to Administer > Customize Data and Screens > Profiles